fabric-entity-attribute-api-v1/src/main/java/net/fabricmc/fabric/impl/entity/attribute/FabricEntityAttributeRegistry.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.entity.attribute;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.attribute.DefaultAttributeContainer;
+import net.minecraft.util.registry.Registry;
+
+import net.fabricmc.fabric.api.entity.attribute.v1.EntityAttributeRegistry;
+
+public final class FabricEntityAttributeRegistry implements EntityAttributeRegistry {
+	public static final FabricEntityAttributeRegistry INSTANCE = new FabricEntityAttributeRegistry();
+	private static final Logger LOGGER = LogManager.getLogger();
+	private final Map<EntityType<? extends LivingEntity>, Supplier<DefaultAttributeContainer.Builder>> pendingRegistrations = new HashMap<>();
+	private Map<EntityType<? extends LivingEntity>, DefaultAttributeContainer> registrations;
+
+	@Override
+	public void register(EntityType<? extends LivingEntity> type, Supplier<DefaultAttributeContainer.Builder> builderSupplier) {
+		if (this.registrations == null) {
```

Comment:
I like the way BlockRenderLayerMapImpl uses BiConsumer's to handle early and late registrations. Its basicly the same as what you have, migt just be a bit cleaner.

Reply:
It may help cut down the number of times the same log is also printed